src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -138,14 +130,11 @@ public async Task<GraphNode<RemoteResolveResult>> WalkAsync(LibraryRange library
             Debug.Assert(node.Item != null, "FindLibraryCached should return an unresolved item instead of null");
 
             // Merge in runtime dependencies
-            if (dependencies?.Count > 0)
+            if (runtimeDependencies?.Count > 0)
             {
                 foreach (var nodeDep in node.Item.Data.Dependencies)
                 {
-                    if (runtimeDependencies?.Contains(nodeDep.Name, StringComparer.OrdinalIgnoreCase) != true)
-                    {
-                        dependencies.Add(nodeDep);
-                    }
+                    runtimeDependencies.Add(nodeDep);
```

Comment:
In the previous implementation, `nodeDep` is added to the `dependencies` list only if it is present in the hashset `runtimeDependencies` already. In the new implementation, we are adding to the hashset directly. I am not sure if there is any untested code path hence an issue in this logic was not caught. Please let me know if I am missing something.

Reply:
dependencies was a list, so it was being done that way to avoid duplicates in dependencies, which is now not possible because dependencies is a hashset